core/src/RIO/Core.hs

```diff
@@ -0,0 +1,70 @@
+-- | This module takes code from RIO.Prelude.RIO and reproduces it here
+--  so that we can propose rio-core package to the rio team.
+--
+{-# OPTIONS -Wno-noncanonical-monoid-instances #-}
+module RIO.Core
```

Comment:
Till we upstream, should we name it `MRIO.Core` ?

Reply:
Yeah, going to carry out this change in the colog package as well

Reply:
Given we are nolonger clashing with `rio` having `MRIO` instead of `MRIO.Core` seems like a good move. What do you think